test/boost/managed_bytes_test.cc

```diff
@@ -122,3 +122,29 @@ SEASTAR_THREAD_TEST_CASE(test_managed_bytes_view_from_bytes) {
     });
     BOOST_REQUIRE_EQUAL(to_bytes(m), b);
 }
+
+SEASTAR_THREAD_TEST_CASE(test_managed_bytes_view_remove_prefix) {
+    bytes b;
+    managed_bytes_view m;
+    size_t n;
+
+    m.remove_prefix(0);
+    BOOST_REQUIRE_EQUAL(to_bytes(m), bytes_view());
+    BOOST_REQUIRE_THROW(m.remove_prefix(1), std::runtime_error);
+
+    b = random_bytes(1, 16);
+    m = managed_bytes_view(b);
+    n = tests::random::get_int(size_t(0), b.size());
+    BOOST_TEST_MESSAGE(format("size={} remove_prefix={}", b.size(), n));
```

Comment:
Random tests should iterate to exercise more cases.

We need a fragmented manged_bytes so the test actually tests something.

Reply:
> Random tests should iterate to exercise more cases.

I wanted to ensure we test both small and large sizes, but doing a number of iterations is possible.

> We need a fragmented manged_bytes so the test actually tests something.

right, will change